### ISSUE TITLE
Suppressing error message at pcm_open

### ISSUE THREAD

**Issue by leuenmat** [tinyalsa/tinyalsa#184]
By returning the pointer to bad_pcm, the previously set error message on pcm with oops(pcm...) is never readable. Calling pcm_get_error will always return an invalid value.

https://github.com/tinyalsa/tinyalsa/blob/16a6e7cd72298af7f1f842e99b3eba53f133c22c/src/pcm.c#L960

**Comment by tay10r** [comment]
Agreed. I'm surprised that this wasn't realized sooner. I think the right course of action would be to always return the PCM pointer, so that the error information is available.
